pkg/antctl/raw/featuregates/command_test.go

```diff
@@ -0,0 +1,393 @@
+		},
+		{
+			"component": "agent",
+			"name": "AntreaProxy",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "agent",
+			"name": "Egress",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "agent",
+			"name": "NodePortLocal",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "agent",
+			"name": "Multicluster",
+			"status": "Enabled",
+			"version": "ALPHA"
+		},
+		{
+			"component": "controller",
+			"name": "AntreaPolicy",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "controller",
+			"name": "NodeIPAM",
+			"status": "Disabled",
+			"version": "ALPHA"
+		},
+		{
+			"component": "controller",
+			"name": "Traceflow",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "controller",
+			"name": "ServiceExternalIP",
+			"status": "Disabled",
+			"version": "ALPHA"
+		},
+		{
+			"component": "controller",
+			"name": "Egress",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "controller",
+			"name": "NetworkPolicyStats",
+			"status": "Enabled",
+			"version": "BETA"
+		}
+	]`)
+
+	agentResponse := []byte(`[
+		{
+			"component": "agent",
+			"name": "EndpointSlice",
+			"status": "Disabled",
+			"version": "ALPHA"
+		},
+		{
+			"component": "agent",
+			"name": "AntreaIPAM",
+			"status": "Disabled",
+			"version": "ALPHA"
+		},
+		{
+			"component": "agent",
+			"name": "FlowExporter",
+			"status": "Disabled",
+			"version": "ALPHA"
+		},
+		{
+			"component": "agent",
+			"name": "NetworkPolicyStats",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "agent",
+			"name": "AntreaPolicy",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "agent",
+			"name": "Multicast",
+			"status": "Disabled",
+			"version": "ALPHA"
+		},
+		{
+			"component": "agent",
+			"name": "Traceflow",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "agent",
+			"name": "ServiceExternalIP",
+			"status": "Disabled",
+			"version": "ALPHA"
+		},
+		{
+			"component": "agent",
+			"name": "AntreaProxy",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "agent",
+			"name": "Egress",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "agent",
+			"name": "NodePortLocal",
+			"status": "Enabled",
+			"version": "BETA"
+		},
+		{
+			"component": "agent",
+			"name": "Multicluster",
+			"status": "Enabled",
+			"version": "ALPHA"
+		}
+	]`)
+
+	k8sClient := k8sfake.NewSimpleClientset(node1.DeepCopyObject())
+	antreaClientset := antreafakeclient.NewSimpleClientset(controllerInfo.DeepCopyObject())
+
+	tests := []struct {
+		name              string
+		runE              func(cmd *cobra.Command, _ []string) error
+		getFakeRestClient func(kubeconfig *rest.Config, k8sClientset kubernetes.Interface, antreaClientset antrea.Interface, mode string) (*rest.RESTClient, error)
```

Comment:
I am not sure why you don't just use the `[]byte` response here instead
you can always do the rest inside the sub-test implementation

Reply:
You are right, refined it, thanks.